paymentsheet/src/test/java/com/stripe/android/customersheet/CustomerSheetViewModelTest.kt

```diff
@@ -46,6 +48,71 @@ class CustomerSheetViewModelTest {
         }
     }
 
+    @Test
+    fun `When payment methods loaded, CustomerSheetViewState is populated`() = runTest {
+        val viewModel = createViewModel(
+            customerAdapter = FakeCustomerAdapter(
+                paymentMethods = Result.success(
+                    listOf(
+                        PaymentMethodFixtures.CARD_PAYMENT_METHOD
+                    )
+                ),
+                selectedPaymentOption = Result.success(
+                    CustomerAdapter.PaymentOption.fromId(
+                        PaymentMethodFixtures.CARD_PAYMENT_METHOD.id!!
+                    )
+                )
+            )
+        )
+        viewModel.viewState.test {
+            assertThat(awaitItem())
+                .isEqualTo(
+                    CustomerSheetViewState.SelectPaymentMethod(
+                        title = null,
+                        paymentMethods = listOf(
+                            PaymentOptionsItem.SavedPaymentMethod(
+                                displayName = "····4242",
+                                paymentMethod = PaymentMethodFixtures.CARD_PAYMENT_METHOD
+                            )
+                        ),
+                        selectedPaymentMethodId = PaymentMethodFixtures.CARD_PAYMENT_METHOD.id!!,
+                        isLiveMode = false,
+                        isProcessing = false,
+                        isEditing = false,
+                        errorMessage = null,
+                    )
+                )
+        }
+    }
+
+    @Test
+    fun `When payment methods cannot be loaded, errorMessage is emitted`() = runTest {
+        val viewModel = createViewModel(
+            customerAdapter = FakeCustomerAdapter(
+                paymentMethods = Result.failure(Exception("Failed to retrieve payment methods."))
+            )
+        )
+        viewModel.viewState.test {
+            assertThat(
+                (awaitItem() as CustomerSheetViewState.SelectPaymentMethod).errorMessage
+            ).isEqualTo("Failed to retrieve payment methods.")
+        }
+    }
+
+    @Test
+    fun `When the selected payment method cannot be loaded, errorMessage is emitted`() = runTest {
```

Comment:
Is this the behavior we want? Should we just revert back to nothing selected?

Reply:
Hmm, yea I think you're right.